metricbeat/module/system/process_summary/process_summary.go

```diff
@@ -68,9 +69,10 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // descriptive error must be returned.
 func (m *MetricSet) Fetch(r mb.ReporterV2) error {
 
-	procList, err := process.ListStates(m.sys)
-	if err != nil {
-		return fmt.Errorf("error fetching process list: %w", err)
+	procList, degradeErr := process.ListStates(m.sys)
```

Comment:
[Suggestion]
Is there any special reason to rename `err` to `degradeErr`? Below, at first it seems unclear why `degradeErr` instead of `err`. I'd suggest to keep them all as `err`

Reply:
Yes, if you look at 
https://github.com/elastic/beats/blob/dd35df4c56e3dc6c0277e039f6dd1d304ade0061/metricbeat/module/system/process_summary/process_summary.go#L88-L91

This part uses the same i.e. `err` and it would overwrite the previous error (which is needed while returning).